x-pack/plugins/ml/public/application/timeseriesexplorer/components/series_controls/series_controls.tsx

```diff
@@ -53,19 +53,36 @@ export type UiPartitionFieldsConfig = Exclude<PartitionFieldsConfig, undefined>;
 
 export type UiPartitionFieldConfig = Exclude<PartitionFieldConfig, undefined>;
 
+/**
+ * Returns a filterBy object based on the partition_field value.
+ */
+const getFilterBy = (entities: Entity[]): Pick<UiPartitionFieldConfig, 'filterBy'> | null => {
```

Comment:
hey @rbrtj, I reckon we should support cross filter here, i.e. if the user select the `by` field first, we should filter out the `partition_field` dropdown accordingly.